### PR TITLE
Compute prognostic run diags against segmented run

### PR DESCRIPTION
**Pull request by nbren12**
For emulation, it is nice to use an existing prognostic run as the verification data. 

This PR adds a `--verification-url` flag to `prognostic_run_diags save` to support this use case and refactors the data loading to a more extensible design.

Added public API:
- new flag: "prognostic_run_diags save --verification-url URL"

Significant internal changes:
- Refactor data loading so that verification and prognostic run data are both instances of the same class. This allows easier plug and play capability.

- [x] Tests added

### CODE REVIEW

**Comment by nbren12** [comment]
It's probably best to review commit by commit.

File: workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/compute.py [hunk 1]
```diff
@@ -399,16 +400,22 @@ def compute_histogram_bias(prognostic, verification, grid):
...
 def register_parser(subparsers):
-    parser = subparsers.add_parser("save", help="Compute the prognostic run diags.")
+    parser: ArgumentParser = subparsers.add_parser(
+        "save", help="Compute the prognostic run diags."
+    )
     parser.add_argument("url", help="Prognostic run output location.")
     parser.add_argument("output", help="Output path including filename.")
     parser.add_argument("--catalog", default=vcm.catalog.catalog_path)
-    parser.add_argument(
+    verification_group = parser.add_mutually_exclusive_group()
```
Comment (AnnaKwa) [inline workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/compute.py:409]: Neat, I didn't know about this functionality.

Comment (AnnaKwa) [reply workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/compute.py:409]: Could you add to the help for the verification options saying that they are mutually exclusive so this appears in the docs?

Comment (AnnaKwa) [reply workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/compute.py:409]: Also, does group cause defaults get ignored if the other flag  is provided? i.e. if `--verification-url` is given, will it ignore the default of `--verification 40day_may2020`?

**Review by AnnaKwa — APPROVED** [review]
LGTM! The refactor of the data loading makes the code a lot tidier!

Can you add some comments to the docstrings so people know about the two mutually exclusive flags?

Also I just wanted to make sure that the mutually exclusive group causes the defaults to be ignored if another flag is set.

File: workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/load_run_data.py [hunk 2]
```diff
@@ -202,22 +203,51 @@ def _insert_nan_from_other(self: xr.Dataset, other: xr.Dataset):
...
+        return load_verification(self._verif_entries["dycore"], self.catalog)
 
+    @property
+    def data_3d(self):
+        return load_verification(self._verif_entries["3d"], self.catalog)
+
+
+def load_verification_and_input_data(url, catalog, verification: str):
+
+    verification = CatalogSimulation(url, verification, catalog)
```
[the remark this answers is not in the document]
Comment (AnnaKwa) [reply workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/load_run_data.py]: Never mind, changed in later commit.

[the remark this answers is not in the document]
Comment (nbren12) [reply workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/load_run_data.py]: This was fixed in e1781275974fa04232055d078e797a2c9d6087a8

File: workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/compute.py (hunk 1 above)
Comment (nbren12) [reply workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/compute.py:409]: Re. defaults. argparse does not do anything special apart from raising a helpful error if both flags are passed. The value of `args.verification` will still be the default value, so I implement the switching logic here: https://github.com/VulcanClimateModeling/fv3net/blob/96426df0c6ee83813f786c0305b3bfd29f02aaef/workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/compute.py#L432-L433.

In terms of documentation, I added some explanation to the help string of `--verification-url` and the usage does indicate these options are mutually exclusive:
```
usage: prognostic_run_diags save [-h] [--catalog CATALOG]
                                 [--verification VERIFICATION | --verification-url VERIFICATION_URL]
                                 [--n-jobs N_JOBS]
                                 url output
```

**Pull request merged**